deckhouse-controller/pkg/controller/moduleloader/loader.go

```diff
@@ -0,0 +1,486 @@
+
+	for _, module := range l.modules {
+		result = append(result, module.GetBasicModule())
+	}
+
+	return result, nil
+}
+
+// LoadModule implements the module loader interface from addon-operator, it reads single directory and returns BasicModule
+// modulePath is in the following format: /deckhouse-controller/downloaded/<module_name>/<module_version>
+func (l *Loader) LoadModule(_, modulePath string) (*modules.BasicModule, error) {
+	if _, err := readDir(modulePath); err != nil {
+		return nil, err
+	}
+
+	// run moduleDefinitionByDir("<module_name>", "/deckhouse-controller/downloaded/<module_name>/<module_version>")
+	def, err := l.moduleDefinitionByDir(filepath.Base(filepath.Dir(modulePath)), modulePath)
+	if err != nil {
+		return nil, err
+	}
+
+	module, err := l.processModuleDefinition(def)
+	if err != nil {
+		return nil, err
+	}
+	l.modules[def.Name] = module
+
+	return module.GetBasicModule(), nil
+}
+
+func (l *Loader) processModuleDefinition(def *Definition) (*Module, error) {
+	if err := validateModuleName(def.Name); err != nil {
+		return nil, err
+	}
+
+	// load values for the module
+	valuesModuleName := utils.ModuleNameToValuesKey(def.Name)
+	// 1. from static values.yaml inside the module
+	moduleStaticValues, err := utils.LoadValuesFileFromDir(def.Path)
+	if err != nil {
+		return nil, err
+	}
+
+	if moduleStaticValues.HasKey(valuesModuleName) {
+		moduleStaticValues = moduleStaticValues.GetKeySection(valuesModuleName)
+	}
+
+	// 2. from openapi defaults
+	configBytes, vb, err := utils.ReadOpenAPIFiles(filepath.Join(def.Path, "openapi"))
+	if err != nil {
+		return nil, err
+	}
+
+	module, err := newModule(def, moduleStaticValues, configBytes, vb, l.log)
+	if err != nil {
+		return nil, err
+	}
+
+	// load conversions
+	if _, err = os.Stat(filepath.Join(def.Path, "openapi", "conversions")); err == nil {
+		l.log.Debugf("conversions for the '%s' module found", def.Name)
+		if err = conversion.Store().Add(def.Name, filepath.Join(def.Path, "openapi", "conversions")); err != nil {
+			l.log.Debugf("failed to load conversions for the '%s' module: %v", def.Name, err)
+			return nil, err
+		}
+	} else {
+		if !os.IsNotExist(err) {
+			l.log.Debugf("failed to load conversions for the '%s' module: %v", def.Name, err)
+			return nil, err
+		}
+		l.log.Debugf("conversions for the '%s' module not found", def.Name)
+	}
+
+	// load constrains
+	if err = extenders.AddConstraints(def.Name, def.Requirements); err != nil {
+		return nil, err
+	}
+
+	return module, nil
+}
+
+func validateModuleName(name string) error {
+	// check if name is consistent for conversions between kebab-case and camelCase.
+	restoredName := utils.ModuleNameFromValuesKey(utils.ModuleNameToValuesKey(name))
+
+	if name != restoredName {
+		return fmt.Errorf("'%s' name should be in kebab-case and be restorable from camelCase: consider renaming to '%s'", name, restoredName)
+	}
+
+	return nil
+}
+
+func (l *Loader) GetModuleByName(name string) (*Module, error) {
```

Comment:
```suggestion
var ErrModuleIsNotFound = errors.New("module is not found")

func (dml *DeckhouseController) GetModuleByName(name string) (*models.DeckhouseModule, error) {
	module, ok := dml.deckhouseModules[name]
	if !ok {
		return nil, ErrModuleIsNotFound
	}

	return module, nil
}
```

we lost named errors

Reply:
fixed